base.go

```diff
@@ -142,15 +142,17 @@ func (db *baseDB) withConn(
 	if ctx != nil && ctx.Done() != nil {
 		fnDone = make(chan struct{})
 		go func() {
-			select {
-			case <-fnDone: // fn has finished, skip cancel
-			case <-ctx.Done():
-				err := db.cancelRequest(cn.ProcessID, cn.SecretKey)
-				if err != nil {
-					internal.Logger.Printf(ctx, "cancelRequest failed: %s", err)
+			for {
```

Comment:
Why is this in a for loop?  I looks like the previous logic would work. This appears to have this for loop run forever, as there is no way to break out of it.

Reply:
Very good point! My bad. Reverted the change.